bodhi/server/models.py

```diff
@@ -2866,6 +2866,8 @@ def set_request(self, db, action, username):
             self.add_tag(self.release.pending_signing_tag)
         elif action is UpdateRequest.stable:
             self.add_tag(self.release.pending_stable_tag)
+            if self.request is UpdateRequest.testing:
+                self.remove_tag(self.release.pending_testing_tag)
```

Comment:
We should check this really happens in tests. You can use [this test](https://github.com/fedora-infra/bodhi/blob/23505cf56ff64b83a65da7ffb728e5976a6afd31/bodhi/tests/server/test_models.py#L3338).

I think you'll need to use `mock.assert_called_with()`... take your time to learn how to use the mock library.

Reply:
ah, and as pointed out by lgtm-bot comment, use `==` instead of `is` ;-)

Reply:
Hi! Sorry for the late response. 

So we need to check if we encountered a case where an update went from pending to stable directly (due to fulfillment of the karma criteria). This is done using the assert statement on the test you gave. If this is true, then we know for sure that the action required for the update to go to stable but the request was to go to testing, i.e. the update is in the pending state wanting to go stable. Hence, the pending-testing tag is needed to be removed.

Could you please help me out with this? 

```
from unittest import mock
from mock import Mock
#to line 3351
mock = Mock()
mock.set_request.assert_called_once_with(self=self.obj, db=self.db , action=UpdateRequest.stable, username=req.user.name)
```

I ran the test using `pytest-3 --no-cov --disable-warnings bodhi/tests/server/test_models.py::TestUpdate::test_set_request_pending_stable` and `AssertionError: Expected 'set_request' to be called once. Called 0 times.` appeared.